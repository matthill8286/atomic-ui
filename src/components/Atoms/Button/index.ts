export * from './Button'
export * from './Button.interface'
export * from './IconButton'
