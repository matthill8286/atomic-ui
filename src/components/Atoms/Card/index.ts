export * from './CardV2'
export * from './Card.interface'
export { StyledCard } from './Card.styled'
