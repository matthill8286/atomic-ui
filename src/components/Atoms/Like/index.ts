export * from './Like'
export * from './Like.interface'
