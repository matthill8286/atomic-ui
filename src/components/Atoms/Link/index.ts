export * from './Link'
export * from './Link.interface'
