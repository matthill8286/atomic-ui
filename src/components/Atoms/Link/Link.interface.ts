import * as H from 'history'
import React, { ComponentType, MouseEvent, MouseEventHandler, ReactElement, ReactNode } from 'react'
import { ThemeColors, ThemeFontLineHeight, ThemeFontSizes } from '@/types/theme'
import { ButtonType } from '@/components/Atoms/Button'

export type LinkTarget = string | undefined

export type LinkHref = string | undefined

export interface WrappedRouterLinkProps {
  children: ReactNode
  className?: string
  onClick?: (e: MouseEvent<HTMLElement>) => void
  onMouseDown?: (e: MouseEvent) => void
  to?: string
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type LinkAs = string | JSX.IntrinsicElements | ComponentType<any>
export type LinkSize = 'small' | 'large'
type CursorType = 'initial' | 'pointer'

export interface LinkProps
  extends LinkRouterProps<HTMLAnchorElement | HTMLButtonElement>,
    React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement> {
  bold?: boolean
  cursor?: CursorType
  color?: ThemeColors
  decorationColor?: ThemeColors
  fontSize?: ThemeFontSizes
  iconLeft?: ReactElement
  iconRight?: ReactElement
  inline?: boolean
  lineHeight?: ThemeFontLineHeight
  scale?: LinkSize
  scrollOffset?: number
  smooth?: boolean
  underline?: boolean
  disabled?: boolean
  type?: ButtonType
  target?: LinkTarget
  href?: LinkHref
  branded?: boolean
  ariaLabel?: string
  isGrouped?: boolean
}

export interface LinkRouterProps<T> {
  to?: H.LocationDescriptor
  children?: React.ReactNode
  className?: string
  onClick?: MouseEventHandler<T>
}

export interface StyledLinkProps {
  color?: ThemeColors
  cursor?: CursorType
  hasIconLeft: boolean
  hasIconRight: boolean
  isBold: boolean
  isGrouped?: boolean
  isDisabled: boolean
  isInline: boolean
  isLarge: boolean
  isSmall: boolean
  isUnderlined?: boolean
  target?: string
  title?: string
}
