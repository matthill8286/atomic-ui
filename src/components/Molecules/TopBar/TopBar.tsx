import * as React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { Lists } from '@/components/Atoms/Lists'
import { Section } from '@/components/Atoms/Section'
import { CopyText } from '@/components/Atoms/Typography'
import { Cell, Grid, Row } from '@/components/Helper/Grid'
import { media } from '@/styles/media'
import { styled } from '@/styles/styled'
import { IconCheckmarkCircle } from '@matthill8286/atomic-icon-library'

interface TopBarProps {
  entries: React.ReactChild[]
}

const StyledLists = styled(Lists)`
  display: flex;
  flex-direction: row;
  z-index: 1;

  > li {
    flex: 1;
    display: none;
    justify-content: center;
    padding: ${({ theme }) => theme.spacing.base.xxs} 0;
    margin: 0 ${({ theme }) => theme.spacing.base.xxs};
    overflow: hidden;

    > span {
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;

      > p {
        text-overflow: ellipsis;
        white-space: nowrap;
        overflow: hidden;
      }
    }

    &:nth-child(-n + 2) {
      display: flex;
    }

    ${media.lg} {
      &:nth-child(-n + 4) {
        display: flex;
      }
    }
  }
`

const StyledIcon = styled(Icon)`
  svg {
    position: relative;
    top: 5px;
  }
`

const StyledSection = styled(Section)`
  background: ${({ theme }) => theme.header.topBar};
`

export const TopBar: React.FC<TopBarProps> = ({ entries }) => {
  return (
    <StyledSection
      color="white"
      paddingTop={{
        mobile: 'xs',
        tablet: 'xs',
      }}
      paddingBottom={{
        mobile: 'xs',
        tablet: 'xs',
      }}>
      <Grid>
        <Row noMargin>
          <Cell colsXs={4} colsSm={8} colsMd={8} colsLg={12} colsXl={12}>
            <StyledLists icon>
              {entries.map((entry, index) => (
                <li key={index}>
                  <StyledIcon height={14} width={14} color="grey5">
                    <IconCheckmarkCircle />
                  </StyledIcon>
                  <CopyText color="grey5" bold fontSize="sm" tag="span">
                    {entry}
                  </CopyText>
                </li>
              ))}
            </StyledLists>
          </Cell>
        </Row>
      </Grid>
    </StyledSection>
  )
}
