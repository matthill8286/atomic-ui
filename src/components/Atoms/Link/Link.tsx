import React from 'react'
import { Link as RouterLink } from 'react-router-dom'
import { smoothScrollWithEvent } from '@/utils'
import { LinkProps, StyledLinkProps } from './Link.interface'
import {
  StyledIconWrapper,
  StyledLinkAnchor,
  StyledLinkButton,
  StyledLinkRouter,
  Text,
} from './Link.styled'

export const Link: React.FC<LinkProps> = ({
  bold = false,
  branded = false,
  children,
  className,
  cursor,
  color = 'grey4',
  decorationColor,
  disabled = false,
  fontSize,
  href,
  iconLeft,
  iconRight,
  inline = false,
  lineHeight,
  onClick,
  onMouseDown,
  scale = 'small',
  scrollOffset = 0,
  smooth = false,
  target = '_self',
  title,
  to,
  underline = true,
  isGrouped = false,
  ariaLabel,
  ...otherProps
}): JSX.Element => {
  const isSmall = scale === 'small'
  const isButton = onClick && !href && !to
  const isRouterLink = to

  const styleProps: StyledLinkProps = {
    color: decorationColor || color,
    cursor,
    hasIconLeft: !!iconLeft,
    hasIconRight: !!iconRight,
    isBold: bold,
    isDisabled: disabled,
    isInline: inline,
    isLarge: scale === 'large',
    isSmall,
    isUnderlined: underline,
    isGrouped,
  }
  let onClickWrapper

  if (smooth) {
    onClickWrapper = (event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>) => {
      try {
        smoothScrollWithEvent(event, scrollOffset)
      } catch (e) {
        //failsafe
      }
      if ('function' == typeof onClick) {
        return onClick(event)
      }
    }
  } else {
    onClickWrapper = onClick
  }

  const getChildren = (): JSX.Element => (
    <>
      {iconLeft && <StyledIconWrapper {...styleProps}>{iconLeft}</StyledIconWrapper>}
      {typeof children === 'string' ? (
        <Text
          tag="span"
          fontSize={fontSize || (isSmall ? 'xs' : 'sm')}
          lineHeight={lineHeight}
          bold={bold}
          color={color}
          isBranded={branded}
          isInline={inline}
          aria-label={ariaLabel}>
          {children}
        </Text>
      ) : (
        children
      )}
      {iconRight && <StyledIconWrapper {...styleProps}>{iconRight}</StyledIconWrapper>}
    </>
  )

  const linkType = (): JSX.Element => {
    if (isButton) {
      return (
        <StyledLinkButton
          className={className}
          {...(disabled ? {} : { onClick: onClickWrapper })}
          {...styleProps}
          {...otherProps}>
          {getChildren()}
        </StyledLinkButton>
      )
    }

    if (isRouterLink) {
      return (
        <StyledLinkRouter
          onClick={onClickWrapper}
          to={to}
          className={className}
          {...styleProps}
          data-test="ewb-router-link">
          {getChildren()}
        </StyledLinkRouter>
      )
    }

    return (
      <StyledLinkAnchor
        className={className}
        target={target}
        rel={target === '_blank' ? 'noopener noreferrer' : otherProps.rel}
        {...(disabled ? {} : { href, onClick: onClickWrapper })}
        {...styleProps}
        {...otherProps}>
        {getChildren()}
      </StyledLinkAnchor>
    )
  }

  return linkType()
}
