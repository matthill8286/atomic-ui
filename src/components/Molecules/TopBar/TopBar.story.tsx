import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { Link } from '@/components/Atoms/Link'
import { TopBar } from './TopBar'

const stories = storiesOf('Design System/Molecules/TopBar', module)

const entries = [
  'Virtual Gathering',
  'Conference Zone',
  'Recommended Assets',
  <Link
    href="www.google.com"
    inline
    bold
    fontSize="sm"
    color="grey5"
    decorationColor="grey5"
    key="link">
    Test
  </Link>,
]

stories.add('Default', () => {
  return <TopBar entries={entries} />
})
