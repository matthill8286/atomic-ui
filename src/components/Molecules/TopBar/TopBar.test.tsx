import React from 'react'
import { Link } from '@/components/Atoms/Link'
import { renderWithTheme } from '@/testRenderer'
import { TopBar } from './TopBar'

const entries = [
  'Versandkostenfrei',
  '0% Finanzierung',
  'Abholung im Markt',
  <Link href="www.google.de" inline color="white" decorationColor="white" key="link">
    Test
  </Link>,
]

describe('TopBar', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<TopBar entries={entries} />)
    expect(tree).toMatchSnapshot()
  })
})
