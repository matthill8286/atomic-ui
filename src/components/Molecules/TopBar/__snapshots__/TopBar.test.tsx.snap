// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
.c10 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c12 {
  color: #FFFFFF;
  font-size: 14px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: inherit;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  border-bottom: 1px solid;
  border-bottom-color: #FFFFFF;
  padding: 0;
}

.c11::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -2px;
  height: 1px;
  left: 0;
  width: calc(100%);
  background-color: #FFFFFF;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c11:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c13 {
  display: inline;
  color: #FFFFFF;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 svg {
  width: 14px;
  height: 14px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #666666;
}

.c8 svg g,
.c8 svg p {
  fill: #666666;
}

.c5 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  list-style-type: none;
}

.c5 li {
  margin: 0;
  padding: 0;
  margin-bottom: 8px;
}

.c5 li p:first-child {
  margin-top: 0;
}

.c5 li p:last-child {
  margin-bottom: 0;
}

.c5 .c7 {
  float: left;
  position: relative;
  top: -2px;
  margin-right: 8px;
}

.c0 {
  width: 100%;
  left: 0;
  right: 0;
  background: #FFFFFF no-repeat center center;
  background-size: cover;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
}

.c2 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  margin: 0;
  margin-bottom: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  z-index: 1;
}

.c6 > li {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4px 0;
  margin: 0 4px;
  overflow: hidden;
}

.c6 > li > span {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c6 > li > span > p {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c6 > li:nth-child(-n + 2) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 svg {
  position: relative;
  top: 5px;
}

.c1 {
  background: white;
}

@media (min-width:752px) {
  .c5 {
    font-size: 16px;
  }

  .c5 .c7 {
    top: 0;
  }
}

@media (min-width:752px) {
  .c0 {
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media (min-width:1008px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@supports (display:grid) {
  .c4 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c4 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c4 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c4 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c4 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1232px) {
  .c4 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media (min-width:512px) {
  .c2 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c2 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c2 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c2 {
    padding: 0 64px;
  }
}

@supports (display:grid) {
  .c3 {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: repeat(4,minmax(0,1fr));
  }
}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@media (min-width:1008px) {
  .c6 > li:nth-child(-n + 4) {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<section
  className="c0 c1"
  color="white"
>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <ul
          className="c5 c6"
        >
          <li>
            <div
              className="c7 c8 c9"
              color="grey5"
              height={14}
              width={14}
            >
              <svg
                viewBox="0 0 80 80"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M25.862 56.46L72.38 7 80 15.084 25.863 72.516 0 45.079l7.502-8.095z"
                />
              </svg>
            </div>
            <span
              className="c10"
              color="grey5"
              fontSize="sm"
            >
              Versandkostenfrei
            </span>
          </li>
          <li>
            <div
              className="c7 c8 c9"
              color="grey5"
              height={14}
              width={14}
            >
              <svg
                viewBox="0 0 80 80"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M25.862 56.46L72.38 7 80 15.084 25.863 72.516 0 45.079l7.502-8.095z"
                />
              </svg>
            </div>
            <span
              className="c10"
              color="grey5"
              fontSize="sm"
            >
              0% Finanzierung
            </span>
          </li>
          <li>
            <div
              className="c7 c8 c9"
              color="grey5"
              height={14}
              width={14}
            >
              <svg
                viewBox="0 0 80 80"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M25.862 56.46L72.38 7 80 15.084 25.863 72.516 0 45.079l7.502-8.095z"
                />
              </svg>
            </div>
            <span
              className="c10"
              color="grey5"
              fontSize="sm"
            >
              Abholung im Markt
            </span>
          </li>
          <li>
            <div
              className="c7 c8 c9"
              color="grey5"
              height={14}
              width={14}
            >
              <svg
                viewBox="0 0 80 80"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M25.862 56.46L72.38 7 80 15.084 25.863 72.516 0 45.079l7.502-8.095z"
                />
              </svg>
            </div>
            <span
              className="c10"
              color="grey5"
              fontSize="sm"
            >
              <a
                className="c11"
                color="white"
                href="www.google.de"
                target="_self"
              >
                <span
                  className="c12 c13"
                  color="white"
                  fontSize="xs"
                >
                  Test
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
